### ISSUE TITLE
Training problems on sky and moving object

### ISSUE THREAD

**Issue by wxlwhu** [nianticlabs/monodepth2#346]
Dear author:
         I am trainning the models on our own dataset (**our data are acquired from a car equiped camera, and the fps is 30 **)and facing the following problems:
        (1) we have changed camera intrinsic and mask areas as described in the corresponding files, however,we still face obvious "holes" in the depth maps.
       (2) Pretrained models (specificly Mono_1024x320 ) are employed the generate depth maps on our own datasets, and the "holes" seldom occur, we are really interested in how this happen.
      (3) we compare the depth maps from our trained and your pretrained models (Mono_1024x320), and it seems that the shapes and outlines in depth map(from our own model) are  somtimes more distinct than those (pretrained Mono_1024x320). However, the consistency in depth maps from our models are generally bad.
# The first img is the test one, and the second is the result from Mono_1024x320, and the last one is the result by our model

![Test](https://user-images.githubusercontent.com/84386569/118649590-3f8e1700-b816-11eb-925d-1be4dd5b9ec4.png)
![Mono1024x320_disp](https://user-images.githubusercontent.com/84386569/118649642-50d72380-b816-11eb-85ca-1ec0eecd42f4.jpg)
![ours_disp](https://user-images.githubusercontent.com/84386569/118649665-56346e00-b816-11eb-85e0-bf4a59527396.jpg)

**Comment by Beniko95J** [comment]
I think it is not a good idea to train Monodepth2 on a dataset containing many dynamic objects, since there is no specific designs in Monodepth2 to handle dynamic objects except when the object moves at the same velocity of camera.

You may try to remove images containing dynamic objects in your dataset and retrain Monodepth2 again to see whether the performance is improved.

**Comment by wxlwhu** [comment]
> I think it is not a good idea to train Monodepth2 on a dataset containing many dynamic objects, since there is no specific designs in Monodepth2 to handle dynamic objects except when the object moves at the same velocity of camera.
> 
> You may try to remove images containing dynamic objects in your dataset and retrain Monodepth2 again to see whether the performance is improved.

Thank you very much for your meaningful suggestion.  It is amazing that the "holes" seldom appear when we use the pretrained models by the authors. We wonder whether some steps be taken on the dataset  during training.

**Comment by mdfirman** [comment]
I agree with @Beniko95J's answer. 

If your custom training dataset has lots of moving objects then you are going to see more holes.

In Monodepth2 we saw fewer holes mostly because KITTI has fewer moving objects.

**Comment by dexter2406** [comment]
> I agree with @Beniko95J's answer.
> 
> If your custom training dataset has lots of moving objects then you are going to see more holes.
> 
> In Monodepth2 we saw fewer holes mostly because KITTI has fewer moving objects.

Thanks for the comment @mdfirman. I agree that KITTI has less similar-speed objects.
However, from the 2nd pictures above, we could notice the pretrained model can correctly detect objects with similar speed, while the self-trained model failed. I wonder if there is any specific tricks you employed, so that your trained model can do better?

**Comment by mdfirman** [comment]
We did nothing special except automasking. As was mentioned before: KITTI doesn't have as many moving vehicles as some other datasets, so automasking is fine for KITTI but probably not enough for this dataset being used here.

In terms of practical suggestions for what you could do to help on your data, maybe some of these could help: 

* Train from stereo pairs instead of mono sequences
* Watch the training data, and manually remove sequences with lots of moving objects
* Use semantic segmentation to help to identify vehicles. But you don't want to mask out all vehicles – just moving ones.
* Try to identify moving vehicles and mask them out from your loss. You could try to do this e.g. with an optical flow method, but it's not easy
* Use a more up-to-date method for training on sequences with moving vehicles, e.g. https://arxiv.org/abs/2010.16404
